### PR TITLE
add scan here function

### PR DESCRIPTION
**Pull request by Samael27**
## Description
Adds a "scan here" Floating Action Button at bottom center of the map. When clicked/touched, will set the current map center as new search location.

## Motivation and Context
When outside hunting Pokemon you often want to check a few nearby places before deciding which way to go. This is way more convenient when you can just move the map one or two kilometers/miles and tap "scan here" instead of having to type in the name of some location.

## How Has This Been Tested?
Tested using docker.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

This is the updated #16 PR and #1231 PR including the last changes.

### CODE REVIEW

**Comment by LordNeoZ** [comment]
You can also set the scanner to follow your location or drag'n'drop the marker real quick. I don't see the real benefit that separates this from those methods...

**Comment by NGPriest** [comment]
I agree with @LordNeoZ

**Comment by Mr-Groch** [comment]
I like the idea - sometimes dragging on very distant location is not easy, you need to zoom out, drag, zoom in, drag once again to correct. With this PR you can simply zoom in to area of interest and click scan here. +1 from me

**Comment by cbogithub** [comment]
:+1:  for this idea

**Comment by Samael27** [comment]
@LordNeoZ and @NGPriest : yes you can drag and drop but on mobile device it's not very easy. I have to do it 3 or 4 times in a row to place the marker where I want. It's not something that I really need but it's a cool feature I think :)

**Comment by LordNeoZ** [comment]
@Samael27 on mobile i click on the "my location" button and then i activate "scan follows location" for 10 seconds and the marker moves on its own. Said that, i really see the benefit when jumping really long distances, and checking the commit, it's kinda just aesthetic, so it doesn't burden the program really.

**Comment by Mr-Groch** [comment]
@LordNeoZ This is not for moving scan to your location just to move to "some" location when browsing map. I really like this idea.

**Comment by primaxius** [comment]
-1 scan here button

**Comment by Samael27** [comment]
Changes are done ;)
I had to rebase to avoid conflicts but I think everything is ok now.

**Comment by st0nec0ld** [comment]
Plz add this feature .... I use my map only mobile to scan my current location, this would be more easier.

**Comment by LordNeoZ** [comment]
@st0nec0ld that is exactly the situation this PR isn't aimed for. You should be using the "scan my location" feature already existing

**Comment by Samael27** [comment]
@st0nec0ld : I agree with @LordNeoZ I use this feature to move the map where I'm go to and know what I'll find. In my case, I scan a big area and it's very long to scan everything that's why I do that. Or sometimes it's to help my girlfriend which is playing somewhere and need help to find a Pokemon (switching apps may restart pogo and take a very long time so I do it for here). And sometimes I juste want to see pokemons in an area I wanna go, .....

**Comment by st0nec0ld** [comment]
I use the last release version 3.2.0, cant find any "scan my location" button!? 
My scanner dont run 24/7, only when i want to check for mons.  It is inconvenient:
tap on top left corner > tap on location settings >swipe "search follows me" on > waiting for gps > swipe search follows me off > swipe "search" on

**Comment by cbogithub** [comment]
@st0nec0ld this PR not merger

**Comment by Samael27** [comment]
@primaxius : do I need to do something else or everything is fine ?

**Comment by Slivo-fr** [comment]
Please merge this for god sake, it's been there since the creation of this repository !

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -23,6 +23,9 @@
 ## Checklist:
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+<!--  NOTE: In order to check code style locally and avoid having your build rejected by Travis, -->
+<!--  run the following commands before you commit: `flake8 .` and `npm run lint`. Fix any -->
+<!--  issues they point out. Note also that flake's NOQA is disabled on Travis. -->
```
Comment (jnovack) [inline .github/PULL_REQUEST_TEMPLATE.md]: Why is this changed in this PR?

File: docs/basic-install/index.rst [hunk 2]
```diff
@@ -87,7 +87,7 @@ Linux/OSX:
 
 In order to run from a git clone, you must compile the front-end assets with node. Make sure you have node installed for your platform:
 
- * `Windows <https://nodejs.org/dist/v4.4.7/node-v4.4.7-x64.msi>`_
+ * `Windows <https://nodejs.org/dist/v4.5.0/node-v4.5.0-x64.msi>`_
```
Comment (jnovack) [inline docs/basic-install/index.rst]: Peripheral change.  Why is it in this PR?

File: docs/basic-install/linux.rst [hunk 3]
```diff
@@ -10,8 +10,10 @@ You can install the required packages on Ubuntu by running the following command
 
 .. code-block:: bash
 
+  sudo apt-get install -y python python-pip python-dev build-essential git
   curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
-  sudo apt-get install -y python python-pip python-dev nodejs nodejs-legacy build-essential
+  sudo apt-get install -y nodejs
+
```
Comment (jnovack) [inline docs/basic-install/linux.rst]: Peripheral change.  Why is it in this PR?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Slivo-fr) [inline .github/PULL_REQUEST_TEMPLATE.md]: It's the rebase I think, just changes that are already live on the main repository

File: docs/basic-install/index.rst (hunk 2 above)
Comment (Slivo-fr) [inline docs/basic-install/index.rst]: Same there, rebased so it include some non related changes

File: docs/basic-install/linux.rst (hunk 3 above)
Comment (Slivo-fr) [inline docs/basic-install/linux.rst]: Same :D

**Comment by Samael27** [comment]
I can't handle all the changes, I had to rebase to be up to date and now there are a lot of changes not from me. Is you look at my commits, only 2 are mine.
Thanks Slivo for taking much more care of my PR than me ;)

**Comment by cbogithub** [comment]
why doesn't merger this PR ?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jnovack) [inline .github/PULL_REQUEST_TEMPLATE.md]: If they were already in `PokemonGoMap:develop` then they wouldn't be in the PR.

I think something is wonky...

**Comment by Slivo-fr** [comment]
They were not when the PR was issued, however it did not get merged wuick enough and some conflicts showed up. Samael repo has so been rebased with develop and that's why the PR include those commits

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Slivo-fr) [inline .github/PULL_REQUEST_TEMPLATE.md]: They were not when the PR was issued, however it did not get merged wuick enough and some conflicts showed up. Samael repo has so been rebased with develop and that's why the PR include those commits

**Comment by Samael27** [comment]
I close the PR for the #1398 one.
